src/util/mod.rs

```diff
@@ -53,3 +54,57 @@ pub fn make_spirv<'a>(data: &'a [u8]) -> super::ShaderModuleSource<'a> {
     );
     super::ShaderModuleSource::SpirV(words)
 }
+
+/// Utility methods not meant to be in the main API.
+pub trait DeviceExt {
+    /// Creates a [`Buffer`] with data to initialize it.
+    fn create_buffer_init(&self, desc: &BufferInitDescriptor) -> crate::Buffer;
+}
+
+impl DeviceExt for crate::Device {
+    fn create_buffer_init(&self, descriptor: &BufferInitDescriptor<'_, '_>) -> crate::Buffer {
+        let unpadded_size = descriptor.contents.len() as crate::BufferAddress;
+        let padding = crate::COPY_BUFFER_ALIGNMENT - unpadded_size % crate::COPY_BUFFER_ALIGNMENT;
+        let padded_size = padding + unpadded_size;
+
+        let wgt_descriptor = wgt::BufferDescriptor {
+            label: descriptor.label.as_ref().map(|x| Cow::Borrowed(&**x)),
+            size: padded_size,
+            usage: descriptor.usage,
+            mapped_at_creation: true,
+        };
+
+        let mut map_context = crate::MapContext::new(padded_size);
+
+        map_context.initial_range = 0..padded_size;
+        let buffer = crate::Buffer {
+            context: Arc::clone(&self.context),
+            id: crate::Context::device_create_buffer(&*self.context, &self.id, &wgt_descriptor),
+            map_context: parking_lot::Mutex::new(map_context),
+            usage: descriptor.usage,
+        };
+
+        let range =
+            crate::Context::buffer_get_mapped_range_mut(&*self.context, &buffer.id, 0..padded_size);
+        range[0..unpadded_size as usize].copy_from_slice(descriptor.contents);
+        for i in unpadded_size..padded_size {
+            range[i as usize] = 0;
+        }
+
+        buffer.unmap();
+        buffer
+    }
+}
+
+/// Describes a [`Buffer`] when allocating.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct BufferInitDescriptor<'a, 'data> {
```

Comment:
sorry for being pendantic, please use a single `'a` lifetime, no need to have two different lifetimes here

Reply:
Those lifetimes may be disjoint though, wouldn't it be more restrictive to only have one?